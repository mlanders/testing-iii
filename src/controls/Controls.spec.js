import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import 'react-testing-library/cleanup-after-each';

import Controls from './Controls';
const mock = jest.fn();

describe('Controls Component', () => {
	test('matches snapshot', () => {
		const tree = renderer.create(<Controls />);
		expect(tree.toJSON()).toMatchSnapshot();
	});
	test('component does not crash', () => {
		render(<Controls />);
	});

	test('"Close Gate" fires a function ', () => {
		const { getByText } = render(
			<Controls closed={false} locked={false} toggleClosed={mock} />
		);

		const button = getByText(/close gate/i);

		fireEvent.click(button);
		expect(mock).toHaveBeenCalled();
	});
	test('"Open Gate" fires a function ', () => {
		const { getByText } = render(<Controls closed={true} locked={false} toggleLocked={mock} />);

		const button = getByText(/open gate/i);

		fireEvent.click(button);
		expect(mock).toHaveBeenCalled();
	});

	test('"Lock Gate" fires a function ', () => {
		const { getByText } = render(<Controls closed={true} locked={false} toggleLocked={mock} />);

		const button = getByText(/lock gate/i);

		fireEvent.click(button);
		expect(mock).toHaveBeenCalled();
	});
	test('"Unlock Gate" fires a function ', () => {
		const { getByText } = render(<Controls closed={true} locked={true} toggleLocked={mock} />);

		const button = getByText(/unlock gate/i);

		fireEvent.click(button);
		expect(mock).toHaveBeenCalled();
	});
	test('lock gate should be disabled if gate is open', () => {
		const { getByText } = render(<Controls closed={false} locked={false} />);

		const button = getByText(/lock gate/i);

		expect(button).toBeDisabled();
	});
	test('open gate should be disabled if gate is locked', () => {
		const { getByText } = render(<Controls closed={true} locked={true} />);

		const button = getByText(/open gate/i);

		expect(button).toBeDisabled();
	});
});
