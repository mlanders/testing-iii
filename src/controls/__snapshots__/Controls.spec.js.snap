// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Component matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lockBtn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="closeBtn"
  >
    Close Gate
  </button>
</div>
`;
